homeassistant/components/green_planet_energy/sensor.py

```diff
@@ -0,0 +1,166 @@
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{UnitOfEnergy.KILO_WATT_HOUR}",
+        suggested_display_precision=4,
+        value_fn=lambda api, data: api.get_highest_price_today(data),
+    ),
+    GreenPlanetEnergySensorEntityDescription(
+        key="gpe_lowest_price_day",
+        translation_key="lowest_price_day",
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{UnitOfEnergy.KILO_WATT_HOUR}",
+        suggested_display_precision=4,
+        value_fn=lambda api, data: api.get_lowest_price_day(data),
+    ),
+    GreenPlanetEnergySensorEntityDescription(
+        key="gpe_lowest_price_night",
+        translation_key="lowest_price_night",
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{UnitOfEnergy.KILO_WATT_HOUR}",
+        suggested_display_precision=4,
+        value_fn=lambda api, data: api.get_lowest_price_night(data),
+    ),
+    GreenPlanetEnergySensorEntityDescription(
+        key="gpe_current_price",
+        translation_key="current_price",
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{UnitOfEnergy.KILO_WATT_HOUR}",
+        suggested_display_precision=4,
+        # Don't use lambda for current_price, handled in native_value property
+        value_fn=None,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: GreenPlanetEnergyConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the Green Planet Energy sensors."""
+    coordinator = config_entry.runtime_data
+
+    async_add_entities(
+        GreenPlanetEnergySensor(coordinator, description, config_entry)
+        for description in SENSOR_DESCRIPTIONS
+    )
+
+
+class GreenPlanetEnergySensor(
+    CoordinatorEntity[GreenPlanetEnergyUpdateCoordinator], SensorEntity
+):
+    """Representation of a Green Planet Energy sensor."""
+
+    entity_description: GreenPlanetEnergySensorEntityDescription
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: GreenPlanetEnergyUpdateCoordinator,
+        description: GreenPlanetEnergySensorEntityDescription,
+        config_entry: GreenPlanetEnergyConfigEntry,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        # Use fixed unique_id with just the key for predictable entity IDs
+        self._attr_unique_id = description.key
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, config_entry.entry_id)},
+            name="Green Planet Energy",
+            entry_type=DeviceEntryType.SERVICE,
+        )
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the state of the sensor."""
+        # Special handling for current_price to use the current hour
+        if self.entity_description.key == "gpe_current_price":
+            return self.coordinator.api.get_current_price(
+                self.coordinator.data, dt_util.now().hour
+            )
```

Comment:
I am not exactly sure why this needs special handling, it just looks fine to put in the entity description